DGL_testing/gpu_test.py

```diff
@@ -0,0 +1,37 @@
+import dgl_ptm
+import torch
+import os
+import time
+os.environ["DGLBACKEND"] = "pytorch"
+
+# Test pytorch cuda is available
+if not torch.cuda.is_available():
+    raise SystemError('GPU access not available with PyTorch') 
+
+nagents = [50000]
+runtime = []
+for i in nagents:
+    start = time.time()
+    # Create an instance of the model class
+    model = dgl_ptm.PovertyTrapModel(model_identifier=f'test{i}')
+
+    # Set the model and simulation parameters
+    # Using DEFAULT values
+    model.set_model_parameters(default=True)
```

Comment:
This is the old API. After merging #36, the argument `default` is removed. By default, the model uses all configs defined by the config module. I see that `device: cpu` in the config module, you need to change it to `cuda`. Note that `config.yaml `is just an example config file and it is not used anywhere in the source code.

Reply:
We may want cpu to be the default value as not all devices have GPU capabilities. Hence the choice of cpu. Does that make sense now to keep it as device: cpu?